### PR TITLE
An attempt to clarify error message for Arguments needing "rename" flag

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** error message enhancement 

Taking the initiative to use a formulation which makes it is clear that no renaming has been
done. Hoping it is ok, see discussion at issue #2987.

### CODE REVIEW

**Comment by herbelin** [comment]
Who should self-assign? I feel that @gares should be the natural candidate. The component is however with @mattam82 as primary maintainer.

**Comment by Zimmi48** [comment]
It looks like this error message wasn't documented at all. Could you take this as an opportunity to add it to the reference manual?

**Comment by herbelin** [comment]
> Could you take this as an opportunity to add it to the reference manual?

I could for one message but I'm unsure I have the time to report about all error messages possibly produced by `Arguments`. So, I think I'd prefer not to do an exception.

**Comment by Zimmi48** [comment]
> So, I think I'd prefer not to do an exception.

Perfect is the enemy of the good. I don't ask you to do it for all error messages but only for the one which you are changing. And steps by steps, we'll make the documentation better. You're too perfectionist Hugo; I'm a pragmatic.

**Comment by herbelin** [comment]
Well, I'm certainly perfectionist, but certainly also pragmatic (if not, I would certainly not work on a so imperfect beast like Coq). But for error messages, I don't think I would appreciate as a user to read a list of message which is unreliable.

Maybe, it is also that I'm not enough convinced that it is worth to document error messages in a manual way (without synchronization with the actual code).

**Comment by Zimmi48** [comment]
OK, I'm failing to convince you here and of course an automated solution would be better in the long run. The thing is we currently provide an error message index: https://coq.inria.fr/refman/coq-exnindex.html. Users can use it to better understand error messages they get. Currently the index is incomplete. What I propose is just to make it more complete. But anyways, I won't bother you with this any longer.

BTW the test-suite needs to be fixed with your PR.

**Comment by herbelin** [comment]
> Well, "selection" would give the impression that it is a conscious selection. 

That's true, so "partial list" maybe? Or even "courtesy of some contributors", "contributed on a voluntary basis", ...?

> I suggest we (you?) open a new issue about this (as food for thought for the moment).

I opened #7373.

**Comment by Zimmi48** [comment]
I would not change the title for now.
Thanks for opening the issue. I'm now going to hide most of this discussion as "off-topic" using a new feature provided by GitHub. I hope you don't mind. You can undo it if you do.

**Comment by herbelin** [comment]
Test-suite adapted. @gares, do you share my attempt to provide what seems to me an improvement of the error message?

**Review by gares — APPROVED** [review]
I'm not very fond of the number of `,` in the error message, but I'm fine with it.

**Comment by herbelin** [comment]
> I'm not very fond of the number of `,`

Me too actually. Maybe the commas around `here` could be dropped? Don't know what a native speaker would recommend.

**Comment by Zimmi48** [comment]
I would at least remove the one between `e.g.` and `here`.

> Don't know what a native speaker would recommend.

Calling @jfehrle to the rescue ;-)

**Comment by jfehrle** [comment]
I think that's a lot of commas, a little hard to parse and a bit too terse.

It's correct to offset a non-essential phrase like "e.g., here, A into R" with commas before and after.  (https://owl.english.purdue.edu/owl/owlprint/607, Rule 3).  In polished writing I would always put a comma after "e.g." just as you would after "For example".  I think the comma after "here" would be OK if that was the beginning of a sentence (ibid Rule 2)--not clear that rule should apply in this case.

Instead, how about changing from the passive voice to the simpler command form, e.g., one of these:

```
Use the "rename" flag to rename arguments, such as A into R.
Use the "rename" flag to rename arguments, such as (here) A into R.
Use the "rename" flag to rename arguments, such as (in this command) A into R.
```

I don't fully understand the context of the message.  It seems to say the user should add 'rename A into R' somewhere to make the operation succeed.  But if Coq knows exactly how to fix it, why shouldn't Coq just silently do so?  Or make it a warning?

Hope this helps.

**Comment by gares** [comment]
> I don't fully understand the context of the message

`Arguments` can be used to document, assert, or modify. Modification requires an explicit flag.

**Comment by herbelin** [comment]
Thanks for help. Then, what about:
- `Use the "rename" flag to rename arguments, such as here A into R.`
- or `Use the "rename" flag to rename arguments (here to rename A into R).`
- or `Flag "rename" expected to rename A into R.`
- ...
Maybe the third one would have finally my preference. Would that be ok?

**Comment by Zimmi48** [comment]
Yes, the third one as my preference as well.

**Comment by herbelin** [comment]
> Yes, the third one has my preference as well.

Well, let's then consider the third message elected. I pushed a new version. Ready on my side and assigned to @maximedenes. @maximedenes, I don't know your exact schedule, but I hope it'll not take you a lot of time.

**Review by maximedenes — APPROVED** [review]
